src/test/java/teammates/test/cases/storage/InstructorsDbTest.java

```diff
@@ -328,6 +330,7 @@ public void testUpdateInstructorByGoogleId() throws Exception {
         assertEquals("new role", instructorUpdated.role);
         assertFalse(instructorUpdated.isDisplayedToStudents);
         assertEquals("New Displayed Name", instructorUpdated.displayedName);
+        assertTrue(instructorUpdated.hasObserverPrivileges());
```

Comment:
In addition to that, you might want to do a `assertFalse(instructorUpdated.has<original-role>Privileges());` since a more privileged role is likely to return true for `hasObserverPrivileges`.